src/python/pants/jvm/resolve/coursier_fetch_integration_test.py

```diff
@@ -723,3 +724,54 @@ def test_failed_to_fetch_jar_given_packaging_pom(rule_runner: RuleRunner) -> Non
         match=r"Exception: No jar found for org.apache.curator:apache-curator:5.5.0. .*",
     ):
         rule_runner.request(CoursierResolvedLockfile, [reqs])
+
+
+@maybe_skip_jdk_test
+def test_force_version(rule_runner):
+    # first check that force_version=False leads to a different version
```

Comment:
This seems a bit fragile? It depends on the state of the world. Although I suppose the worst thing that happens is that the test fails when it should pass (because force_version=False happens not to lead to a different version). 

I do think that checking the generated command line is a better and more efficient way to test this. But I think we can do that in a followup.